app/src/main/java/com/github/se/orator/model/chatGPT/ChatViewModel.kt

```diff
@@ -516,7 +517,117 @@ The session is now over. According to the initial instructions, you can now brea
     getNextGPTResponse()
   }
 
-  private fun messagesToTranscript(messages: List<Message>): String {
-    return messages.filter { it.role == "user" }.joinToString("\n") { it.content }
+  /**
+   * Evaluates the battle candidates and provides feedback to the winner and loser.
+   *
+   * @param userUid The UID of the user.
+   * @param messages The messages of that user
+   */
+  private fun messagesToTranscript(userUid: String, messages: List<Message>): String {
+    return messages.joinToString("\n") {
+      if (it.role == "user") "$userUid: ${it.content}" else "Interviewer: ${it.content}"
+    }
+  }
+
+  /**
+   * Evaluates the battle candidates, determines the winner of the battle and provides feedback to
+   * the winner and loser.
+   *
+   * @param user1Uid The UID of the first user.
+   * @param user2Uid The UID of the second user.
+   * @param user1messages The messages of the first user.
+   * @param user2messages The messages of the second user.
+   */
+  suspend fun evaluateBattleCandidates(
+      user1Uid: String,
+      user2Uid: String,
+      user1messages: List<Message>,
+      user2messages: List<Message>,
+      user1name: String,
+      user2name: String
+  ): EvaluationResult {
+    val user1Transcript = messagesToTranscript(user1Uid, user1messages)
+    val user2Transcript = messagesToTranscript(user2Uid, user2messages)
+
+    val prompt =
+        """
+        You are simulating a highly strict recruiter deciding between two candidates, $user1Uid and $user2Uid.
+        
+        You have all their Q&A transcripts:
+        
+        $user1Uid's transcript:
+        $user1Transcript
+
+        $user2Uid's transcript:
+        $user2Transcript
+
+        Please determine who performed better and explicitly state: "The winner is $user1Uid" or "The winner is $user2Uid".
+        Then, provide a message addressed to the winner, highlighting why they won (max one paragraph). 
+        Also, provide a message addressed to the loser, highlighting what they did wrong and where they can improve (max one paragraph).
```

Comment:
Maybe a better prompt would be : 

```plaintext
You are an impartial and strict recruiter tasked with evaluating two AI interview candidates, $user1Uid and $user2Uid, based on their interview transcripts. Your goal is to determine which AI performed better and provide constructive feedback for both.

            **Guidelines:**

            1. **Evaluation Criteria:**
               - **Clarity and Coherence:** Assess how clearly and logically each AI communicated their answers.
               - **Depth of Responses:** Evaluate the thoroughness and depth of each AI's answers.
               - **Relevance to Questions:** Determine how well each AI addressed the interview questions.
               - **Problem-Solving Ability:** Analyze the effectiveness of each AI's problem-solving approach.
               - **Professionalism:** Consider the professionalism displayed in each AI's responses.

            2. **Determining the Winner:**
               - Based on the criteria above, decide which AI performed better overall.
               - Explicitly state the winner in the format: "The winner is: <winnerUid>."

            3. **Feedback for Both AIs:**
               - **Winner's Feedback:** Provide specific strengths that led to their victory.
               - **Loser's Feedback:** Offer constructive criticism highlighting areas for improvement.

            **Format your response as follows:**

            ```
            The winner is: <winnerUid>
            Winner's Feedback:
            - <Strength 1>
            - <Strength 2>
            - <Strength 3>

            Loser's Feedback:
            - <Area for Improvement 1>
            - <Area for Improvement 2>
            - <Area for Improvement 3>
            ```
            
            **Example:**

            ```
            The winner is: user1Uid
            Winner's Feedback:
            - Demonstrated exceptional problem-solving skills with clear, logical explanations.
            - Provided comprehensive and relevant answers to all questions.
            - Maintained a high level of professionalism throughout the interview.

            Loser's Feedback:
            - Responses lacked depth and were sometimes vague.
            - Failed to fully address some of the interview questions.
            - Could improve clarity and coherence in communication.
            ```

            **Transcripts:**

            **$user1Uid's Transcript:**
            $user1Transcript

            **$user2Uid's Transcript:**
            $user2Transcript
        """.trimIndent()
    }
}
```

Reply:
Thank you for the suggestion! It's a great improvement compared to my version